etl_framework/config_mixins/UpsertStatementMixin.py

```diff
@@ -1,46 +1,91 @@
-                    }
+        None: 'VALUES(`{0}`)',
+        'new_values': 'COALESE(VALUES(`{0}`), `{0}`)',
+        'old_values': 'COALESCE(`{0}`, VALUES(`{0}`))'
+    }
 
-    def create_on_duplicate_key_update_statement(self, fields, coalesce=None, statement_string=False):
+    @classmethod
+    def _create_on_duplicate_key_update_statement(cls, fields, coalesce=None,
+                                                  statement_string=False):
         """returns update statement for upsert"""
 
-        coalesce_string = self.COALESCE_MAP[coalesce]
+        coalesce_string = cls.COALESCE_MAP[coalesce]
+        sql_obj = SqlClause(header='ON DUPLICATE KEY UPDATE',
+            phrases=['{} = '.format(field) + coalesce_string.format(field) for field in fields])
 
-        if statement_string:
-            return [], SqlClause(header='ON DUPLICATE KEY UPDATE',
-                phrases=['`{}` = '.format(field) + coalesce_string.format(field) for field in fields]).get_sql_clause()
-        else:
-            return [], SqlClause(header='ON DUPLICATE KEY UPDATE',
-                phrases=['`{}` = '.format(field) + coalesce_string.format(field) for field in fields])
+        return sql_obj
 
     def create_upsert_statement(self, table, fields, statement_string=False):
         """returns update statement"""
 
         insert_fields, insert_statement = self.create_insert_statement(table, fields)
-        update_fields, update_statement = self.create_on_duplicate_key_update_statement(fields)
+        update_statement = self._create_on_duplicate_key_update_statement(fields)
+
+        sql = SqlClause(phrases=[insert_statement, update_statement],
+                        phrase_indents=0,
+                        phrase_separator='\n').get_sql_clause()
 
         if statement_string:
-            return insert_fields + update_fields, SqlClause(phrases=[insert_statement, update_statement],
-                                                            phrase_indents=0,
-                                                            phrase_separator='\n').get_sql_clause()
-        else:
-            return insert_fields + update_fields, SqlClause(phrases=[insert_statement, update_statement],
-                                                            phrase_indents=0,
-                                                            phrase_separator='\n')
+            sql = sql.get_sql_clause()
+
+        return insert_fields, sql
 
     def get_upsert_statement(self):
         """returns statement to upsert data"""
 
         return  self.create_upsert_statement(table=self.get_loader_table(),
                                                 fields=self.get_loader_fields(), statement_string=True)
+
+
+class PostgreSqlUpsertStatementMixin(PostgreSqlInsertStatementMixin):
+    """requires LoaderMixin"""
+
+    COALESCE_MAP = {
+        None: 'EXCLUDED.{0}',
+        'new_values': 'COALESE(EXCLUDED.{0}, {0})',
```

Comment:
@ianlofs  we need to have something like `COALESCE(EXCLUDED.field_name, table_name.field_name`.  Not sure the cleanest way to do that...